synapse/rest/client/versions.py

```diff
@@ -27,11 +27,17 @@ class VersionsRestServlet(RestServlet):
     def on_GET(self, request):
         return (200, {
             "versions": [
-                "r0.0.1",
-                "r0.1.0",
-                "r0.2.0",
+                # we don't need to include the minor values any more, as we've defined
+                # X.Y.Z to be backwards-compatible with X.(Y-1).* where Y>1
```

Comment:
Was this the conclusion of that conversation? I thought for r0 we were saying Y was effectively the major version.

Reply:
i was saying this, but on closer inspection it turned out we’d never defined it as such.

Reply:
I think this needs a bit more discussion, which should probably happen outside of the comment box on github.

r0.3.0 isn't compatible with r0.2.0 (push rules changed, some APIs removed, event formats changed) and if I recall correctly r0.3.0 was released intentionally as r0 given the constraints of the API.

Reply:
agreed. @richvdh, wdyt?

Reply:
yes I agree this needs more discussion :/.

For example, presumably the change to push rules strictly means that we *don't* support < r0.2.0 any more.

My inclination would be to avoid poking at this hornet's nest for now.

Reply:
have unpoked.